backend/app/controllers/memos_controller.rb

```diff
@@ -10,7 +10,7 @@ def index
 
   # GET /memos/:id
   def show
-    @memo = Memo.find(params[:id])
+    @memo = Memo.includes(:tags).find(params[:id])
```

Comment:
Nits

includesはeager_loadにもpreloadの挙動にもなり得るので、この場合はpreloadを使用するようにしましょう！
```suggestion
    @memo = Memo.preload(:tags).find(params[:id])
```

Ref: https://tech.stmn.co.jp/entry/2020/11/30/145159#:~:text=%E3%81%AB%E3%81%AA%E3%82%8A%E3%81%BE%E3%81%99%E3%80%82-,preload%E3%80%81eager_load%E3%81%AE%E4%BD%BF%E3%81%84%E5%88%86%E3%81%91,-%E4%B8%8A%E8%BF%B0%E3%81%AE%E5%86%85%E5%AE%B9

---

へえ、直接関連していなくても、has_many throughの記述があれば先読みできるんですね

Reply:
> 結合で一括取得か、テーブル毎にin句で取得か、今回は、関連先の要素で絞り込みを行ったりしないし、preloadのが良いということですかね！

確かに基本的にLEFT JOINすると重複が生まれるためにpreloadの方が速そうに思えますが、
必ずしもそうでないわけではないので、毎回preloadがいいわけではないとは思います！
また、あまりにもIN句が大きすぎるとSQL文のbyte数が大きくなっちゃってのlimitに引っかかることもあります。
Ref: https://moroto1122.hatenadiary.org/entry/20100805/1280992251

Reply:
preloadとeager_loadの速度を比較したいなら、benchmarkとってみるといいと思います。
（「推測するな、計測せよ」の精神です。）
https://qiita.com/scivola/items/c5b2aeaf7d67a9ef310a